crates/red_knot_python_semantic/resources/mdtest/call/constructor.md

```diff
@@ -1,7 +1,325 @@
 # Constructor
 
+When classes are instantiated, Python calls the meta-class `__call__` method, which can either be
+customized by the user or `type.__call__` is used.
+
+The latter calls the `__new__` method of the class, which is responsible for creating the instance
+and then calls the `__init__` method on the resulting instance to initialize it with the same
+arguments.
+
+Both `__new__` and `__init__` are looked up using full descriptor protocol, but `__new__` is then
+called as an implicit static, rather than bound method with `cls` passed as the first argument.
+`__init__` has no special handling, it is fetched as bound method and is called just like any other
+method.
```

Comment:
Minor nitpick:
```suggestion
`__init__` has no special handling, it is fetched as bound method and is called just like any other
dunder method.
```

(Normal methods would also be looked up on the instance dictionary)